src/main/java/duke/Parser.java

```diff
@@ -0,0 +1,154 @@
+package duke;
+
+import duke.command.*;
+import java.time.LocalDate;
+import java.time.format.DateTimeFormatter;
+import java.time.format.DateTimeParseException;
+
+/**
+ * Parser class represents the parser which parses the command given by user
+ * in the terminal.
+ */
+public class Parser {
+    /**
+     * Parses the command given by user in the terminal
+     * and returns the corresponding Command object based
+     * on command given by user.
+     *
+     * @param command Command given by the user in terminal.
+     * @param taskList The tasks stored in the list.
+     * @return The corresponding Command object is returned.
+     * @throws DukeException If valid description or number is missing after command.
+     * @throws DateTimeParseException If the user does not key in valid date.
+     * @throws NumberFormatException If the user does not choose a valid number.
+     */
+    public static Command parse(String command, TaskList taskList) throws DukeException,
+            DateTimeParseException, NumberFormatException{
+        if (command.toLowerCase().equals("bye")) {
+            return new ByeCommand();
+        }
+        if (command.toLowerCase().equals("list")) {
+            return new ListCommand();
+        }
+        if (command.toLowerCase().startsWith("mark")) {
+            int position = parseMarkedTask(command, taskList);
+            return new MarkCommand(position);
+        }
+        if (command.toLowerCase().startsWith("unmark")) {
+            int position = parseUnmarkedTask(command, taskList);
+            return new UnmarkCommand(position);
+        }
+        if (command.toLowerCase().startsWith("todo")) {
+            String description = parseToDo(command);
+            return new ToDoCommand(description);
+        }
+        if (command.toLowerCase().startsWith("deadline")) {
+           String[] taskWithDeadline = parseDeadline(command);
+           LocalDate date = parseDate(taskWithDeadline[1]);
+           return new DeadlineCommand(taskWithDeadline[0], date);
+        }
+        if (command.toLowerCase().startsWith("event")) {
+            String[] taskWithPeriod = parseEvent(command);
+            LocalDate date = parseDate(taskWithPeriod[1]);
+            return new EventCommand(taskWithPeriod[0], date);
+        }
+        if (command.toLowerCase().startsWith("delete")) {
+            int position = parseDelete(command, taskList);
+            return new DeleteCommand(position);
+        }
+        if (command.toLowerCase().startsWith("find")) {
+            String keyword = parseFind(command);
+            return new FindCommand(keyword);
```

Comment:
Consider using switch statement instead of if-else to make the code look cleaner